linode/vpcips/framework_datasource.go

```diff
@@ -57,7 +57,7 @@ func (r *DataSource) Read(
 		result, d = filterConfig.GetAndFilter(
 			ctx, r.Meta.Client, data.Filters,
 			func(ctx context.Context, client *linodego.Client, filter string) ([]any, error) {
-				return ListVPCIPs(ctx, client, filter, int(data.VPCID.ValueInt64()))
+				return ListVPCIPs(ctx, client, filter, helper.FrameworkSafeInt64ToInt(data.VPCID.ValueInt64(), &resp.Diagnostics))
```

Comment:
Since `FrameworkSafeInt64ToInt` may place an error in `resp.Diagnostics`, I think we can convert this ID around line 53-56, directly in the if block and then we can inspect the error there and return earlier as needed. How does that sound?

Reply:
Good call, I think that is a better solution. Thanks!

Reply:
Fixed!